relevanceai/api/endpoints/deployables/dashboard.py

```diff
@@ -0,0 +1,138 @@
+from abc import ABC, abstractmethod
+from typing import Collection
+
+from relevanceai.api.endpoints.datasets.datasets import DatasetsClient
+from relevanceai.base import _Base
+from .deployables import Deployable
+
+
+class Dashboard(ABC, _Base):
+    def __init__(
+        self, project: str, api_key: str, deployable_id: str, application: str
+    ):
+        deployables = Deployable(project, api_key)
+        if deployable_id not in deployables.list():
+            raise ValueError(f"No deployable with ID {deployable_id}")
+        else:
+            configuration = deployables.get(deployable_id)["configuration"]
+            if not application == configuration["type"]:
+                raise ValueError(f"{deployable_id} is not a {application} application")
+            else:
+                self.vector_field = configuration["vector_field"]
+
+        self.project = project
+        self.api_key = api_key
+        self.deployable_id = deployable_id
+        self.shareable_id = None
+
+    @property
+    @abstractmethod
+    def deployable_url(self):
+        pass
+
+    @property
+    @abstractmethod
+    def shareable_url(self):
+        pass
+
+    @classmethod
+    @abstractmethod
+    def create_application(cls, project: str, api_key: str):
+        pass
+
+
+class Clusters(Dashboard):
+    def __init__(self, project: str, api_key: str, deployable_id: str):
+        super().__init__(project, api_key, deployable_id, "cluster")
+
+    @property
+    def deployable_url(self):
+        deployables = Deployable(self.project, self.api_key)
+        configuration = deployables.get(self.deployable_id)["configuration"]
+        url = "https://cloud.relevance.ai/dataset/{}/deploy/{}/{}/{}/{}"
+        return url.format(
+            configuration["dataset_id"],
+            self.project,
+            "cluster",
+            self.api_key,
+            self.deployable_id,
+        )
+
+    @property
+    def shareable_url(self):
+        if self.shareable_id is None:
+            # TODO: write exception
+            raise Exception
+        else:
+            deployables = Deployable(self.project, self.api_key)
+            configuration = deployables.get(self.deployable_id)["configuration"]
+            url = "https://cloud.relevance.ai/dataset/{}/deploy/{}/{}/{}/{}"
```

Comment:
This reminds me that we need to move the `base_url` property in `http_client` into the default transport layer :)

Reply:
Yes, that would be nice to have